### PR TITLE
fetch the deploy strategy from the environment instead of chasing the deploybackend

### PR DESCRIPTION
**Pull request by stojg**

### CODE REVIEW

**Comment by stojg** [comment]
I think the DNEnvironments needs to planning the deploys, in collaboration with it's deployment backend. This is a first step to break the direct dependency between DNRoot -> deployment backend.

File: code/model/DNEnvironment.php [hunk 1]
```diff
@@ -210,6 +210,15 @@ public function Backend() {
 		return Injector::inst()->get($backend);
 	}
 
+	/**
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return DeploymentStrategy
+	 */
+	public function getDeployStrategy(\SS_HTTPRequest $request) {
+		return $this->Backend()->planDeploy($this, $request);;
```
Comment (halkyon) [inline code/model/DNEnvironment.php]: DOUBLE COLON

Comment (stojg) [inline code/model/DNEnvironment.php]: double tap

**Comment by mateusz** [comment]
As you say, sir!

**Pull request merged**